### PR TITLE
Fix ENV based configuration of Net::Http for proxies

### PR DESCRIPTION
**Pull request by bbergstrom**
Net::Http has a slightly odd constructor semantics that aren't
immediately obvious. If no proxy parameters are sent then, by default, it
looks up the proxy from the standard HTTP_PROXY/http+proxy environment
variables. To set no proxy at all, ignoring the environment variable, you can pass nil
as the p_address parameter.

ActiveMerchant::Connection was always passing proxy parameters
to Net::Http.new even when a proxy had not been set, meaning that
the default environment variable configuration behaviour was never respected.

This commit ensures that if you do not explicitly set proxy parameters
on your Gateway the default behaviour of Net::Http to retrieve proxy
settings from the environment applies. Proxy parameters can still explicitly
set to nil to avoid environment variable configuration or to desired proxy values.

Ref: https://ruby-doc.org/stdlib-2.4.1/libdoc/net/http/rdoc/Net/HTTP.html#method-c-new
Fixes: https://github.com/activemerchant/active_merchant/issues/2372
Revision of dormant PR: https://github.com/activemerchant/active_merchant/pull/2371

### CODE REVIEW

File: test/unit/connection_test.rb [hunk 1]
```diff
@@ -26,6 +26,25 @@ def test_connection_endpoint_raises_uri_error
...
+    @connection.logger.expects(:info).twice
+
+    spy = Net::HTTP.new('example.com', 443)
+    Net::HTTP.expects(:new).with('example.com', 443, :ENV, nil).returns(spy)
+    spy.expects(:get).with('/tx.php', {}).returns(@ok)
+    response = @connection.request(:get, nil, {})
+  end
+
+  def test_connection_does_pass_requested_proxy
+    @connection.logger.expects(:info).twice
```
Comment (bdewater) [inline test/unit/connection_test.rb]: This seems unrelated to what we're testing here, can we remove it? Same question for the other test.

File: test/unit/connection_test.rb [hunk 2]
```diff
@@ -26,6 +26,25 @@ def test_connection_endpoint_raises_uri_error
...
+    @connection.proxy_address = "proxy.example.com"
+    @connection.proxy_port = 8080
+    spy = Net::HTTP.new('example.com', 443)
+    Net::HTTP.expects(:new).with('example.com', 443, "proxy.example.com", 8080).returns(spy)
+    spy.expects(:get).with('/tx.php', {}).returns(@ok)
+    response = @connection.request(:get, nil, {})
```
Comment (bdewater) [inline test/unit/connection_test.rb]: Nitpick: we can remove the unused `response ` variable. Same for the other test.

**Comment by bbergstrom** [comment]
Revised.

**Comment by bbergstrom** [comment]
@bdewater How often does a release get cut of this gem?

**Comment by bdewater** [comment]
More-or-less monthly, there's no firm schedule. You could always point your Gemfile to the merged commit on master if you need it now, the delta with the latest release is pretty small :)
